tsdb/chunks/chunks.go

```diff
@@ -66,20 +66,35 @@ type HeadSeriesRef uint64
 // The HeadSeriesRef and ChunkID may not exceed 5 and 3 bytes respectively.
 type HeadChunkRef uint64
 
-func NewHeadChunkRef(hsr HeadSeriesRef, chunkID uint64) HeadChunkRef {
+func NewHeadChunkRef(hsr HeadSeriesRef, chunkID HeadChunkID) HeadChunkRef {
 	if hsr > (1<<40)-1 {
 		panic("series ID exceeds 5 bytes")
 	}
 	if chunkID > (1<<24)-1 {
 		panic("chunk ID exceeds 3 bytes")
 	}
-	return HeadChunkRef(uint64(hsr<<24) | chunkID)
+	return HeadChunkRef(uint64(hsr<<24) | uint64(chunkID))
 }
 
-func (p HeadChunkRef) Unpack() (HeadSeriesRef, uint64) {
-	return HeadSeriesRef(p >> 24), uint64(p<<40) >> 40
+func (p HeadChunkRef) Unpack() (HeadSeriesRef, HeadChunkID) {
+	return HeadSeriesRef(p >> 24), HeadChunkID(p<<40) >> 40
 }
 
+// HeadChunkID refers to a head chunk for a specific series in Head (memSeries).
```

Comment:
This ID in itself does not specify a series, rather a chunk.
```suggestion
// HeadChunkID refers to a specific chunk in a series (memSeries) in the Head.
```